src/NuGet.Services.AzureSearch/VersionList/FilteredVersionList.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Versioning;
+
+namespace NuGet.Services.AzureSearch
+{
+    /// <summary>
+    /// This type tracks a version list for a specific <see cref="SearchFilters"/>. In other words, this implementation
+    /// assumes that a non-applicable version is never added to the list. In practice, <see cref="VersionLists"/> does
+    /// this filtering before adding a version to this list.
+    /// </summary>
+    internal class FilteredVersionList
+    {
+        internal NuGetVersion _latest;
+        internal readonly SortedList<NuGetVersion, FilteredVersionProperties> _versions;
+
+        public FilteredVersionList(IEnumerable<FilteredVersionProperties> versions)
```

Comment:
>versions [](start = 74, length = 8)

This type is internal, but maybe we should add a null check on `versions`?

Reply:
sure